### PR TITLE
Add support for writing column encoding with sequence id

### PR DESCRIPTION
**Pull request by sdruzkin**
This change allows OrcWriter to write DWRF ColumnEncoding objects with sequence ids and DwrfProto.KeyInfo keys set.

This will be used by flat map writer which is going to use column encoding objects instead of using ColumnWriters for map keys.

Test plan:
- added tests verifying conversion of nested ColumnEncodings with numeric and binary keys
- tested with Vader

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by sdruzkin** [comment]
Please hold on on this PR (idk how to put it back to draft). Looks like I'm missing a part for non nested ColumnEncodings with valid zero sequence

**Comment by sdruzkin** [comment]
Rebased. Ran Vader on 640 files - no issues. Good to merge.

**Review by arunthirupathi — APPROVED** [review]
Looks good, just minor nits.

File: presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataWriter.java [hunk 1]
```diff
@@ -323,20 +323,61 @@ static DwrfProto.Stream.Kind toStreamKind(StreamKind streamKind)
         throw new IllegalArgumentException("Unsupported stream kind: " + streamKind);
     }
 
-    public static DwrfProto.ColumnEncoding toColumnEncoding(int columnId, ColumnEncoding columnEncoding)
+    public static List<DwrfProto.ColumnEncoding> toColumnEncodings(Map<Integer, ColumnEncoding> columnEncodingsByNodeId)
+    {
+        ImmutableList.Builder<DwrfProto.ColumnEncoding> columnEncodings = ImmutableList.builder();
+        for (Entry<Integer, ColumnEncoding> entry : columnEncodingsByNodeId.entrySet()) {
+            Integer nodeId = entry.getKey();
```
[the remark this answers is not in the document]
Comment (sdruzkin) [reply presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataWriter.java]: Fixed

File: presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataWriter.java [hunk 2]
```diff
@@ -323,20 +323,61 @@ static DwrfProto.Stream.Kind toStreamKind(StreamKind streamKind)
...
+            ColumnEncoding columnEncoding = entry.getValue();
+
+            if (columnEncoding.getAdditionalSequenceEncodings().isPresent()) {
+                SortedMap<Integer, DwrfSequenceEncoding> sequences = columnEncoding.getAdditionalSequenceEncodings().get();
```
[the remark this answers is not in the document]
Comment (sdruzkin) [reply presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataWriter.java]: Changed to Map.

**Pull request merged**